### ISSUE TITLE
Everything is discretized: Networking will fail if this is true

### ISSUE THREAD

**Issue by jceipek** [jceipek/Complete-Galactic-Dominion#16]
Decisions made by a unit based on its surroundings currently occur based on a discrete unit state. For example, a unit must check to see if it is in range before it attacks. In order for a deterministic system to be implemented, units must be able to make these kind of checks based on their "true" positions between discrete steps, using a timestamp.

I propose a function that will determine the "true" position of a unit at a given time to allow this to be done.

**Comment by pvarin** [comment]
Agreed. I've been saying this from the start. However, right now (i.e. before demo day) although deep down I know that the client behavior is not deterministic, because the clients are receiving the requests from the server at approximately the same time, their behavior appears simultaneous and therefore deterministic.
I vote to address this issue once Soft Design is over.
